lib/battle.rb

```diff
@@ -0,0 +1,65 @@
+class Battle
+
+  attr_reader :computer_health,
+              :user_health,
+              :computer_board,
+              :user_board,
+              :ships
+
+  def initialize
+    @computer_health = 0
+    @user_health = 0
+    @computer_board = Board.new
+    @user_board = Board.new
+    @ships = [Ship.new("Cruiser", 3), Ship.new("Submarine", 2)]
+  end
+
+  def welcome
+    return "Welcome to BATTLESHIP\nEnter p to play. Enter q to quit."
+    user_input = gets.chomp
+    if user_input == p 
+      set_up
+    elsif user_input == q
+      quit
+    end
```

Comment:
We should also throw in an else branch to account for the user writing anything other than p or q.
We can get funny with it for some personality; like "Don't be cheeky, just type p or q please!"

Optional: We could add in some funny extra alternatives: 
user inputs "play" ---> "I only asked you to put p. But fine, go above and beyond. Ready? You're gonna get SUNK"
user inputs "quit" ---> "Fine, be that way, you only had to write 'q' you know."

We can also account for: user inputs capital P or capital Q. We could make it accept either lowercase or uppercase by throwing in an `.upcase` or `.downcase` somewhere. Or we could make the computer give another sarcastic line.